src/Client.php

```diff
@@ -399,11 +471,23 @@ public function queueMessage(Message $message)
    */
   public function fire($job, array $data)
   {
-    $encoded = json_encode($data, JSON_UNESCAPED_UNICODE);
-
-    $this->guzzle->post($this->endpoint, ['body' => $encoded]);
-
-    $job->delete();
+    if($job->attempts() >= $data['metadata']['num_retries'])
```

Comment:
Does $job->attempts variable get incremented after a job->release() or does it get incremented as soon as fire is called? If it's the latter, this will be a problem. Though, I am pretty sure that it's the former. Either way, let's just test this once?